pluginClientSdkLib/src/main/java/com/twofortyfouram/locale/sdk/client/receiver/AbstractAsyncReceiver.java

```diff
@@ -62,6 +62,7 @@
                 new Pair<PendingResult, AsyncCallback>(pendingResult, callback)));
     }
 
+    @SuppressWarnings("unchecked")
```

Comment:
That feels a bit over-engineered to me - if you don't mind the lint
warning, I will just delete the @SuppressWarnings("unchecked").

J

On Sun, Mar 15, 2015 at 4:42 PM Carter Jernigan <notifications@github.com>
wrote:

> In
> pluginClientSdkLib/src/main/java/com/twofortyfouram/locale/sdk/client/receiver/AbstractAsyncReceiver.java
> <https://github.com/twofortyfouram/android-plugin-client-sdk-for-locale/pull/1#discussion_r26452085>
> :
>
> > @@ -62,6 +62,7 @@
> >                  new Pair<PendingResult, AsyncCallback>(pendingResult, callback)));
> >      }
> >
> > +    @SuppressWarnings("unchecked")
>
> I usually try limit the scope of a suppress annotation as much as
> possible, so that future warnings don't get accidentally suppressed. What
> about a small helper method to perform the cast, and only suppress warnings
> for that helper method? For example:
>
>     @NonNull
>     @SuppressWarnings("unchecked")
>     private static Pair<PendingResult, AsyncCallback> castObj(@NonNull final Object o) {
>         return (Pair<PendingResult, AsyncCallback>) o;
>     }
>
> Then the Handler could call:
>
>     final Pair<PendingResult, AsyncCallback> pair = castObj(msg.obj);
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twofortyfouram/android-plugin-client-sdk-for-locale/pull/1/files#r26452085>
> .
>